### PR TITLE
Usbd-ccid: fix panic on x86_64

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
The length of `usize::to_be_bytes` depends on the size of the pointer on the targeted cpu architecture

This PR also runs cargo fmt.

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
@nickray this might interest you. We ran into it trying to have a usbip runner.

File: components/usbd-ccid/src/types/tlv.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (robin-nitrokey) [inline components/usbd-ccid/src/types/tlv.rs:1]: Typo?  Anyway, can we delete this module?

Comment (sosthene-nitrokey) [reply components/usbd-ccid/src/types/tlv.rs:1]: I think it's cargo fmt. I don't remember opening this file.
I'll remove it.

I noticed there's also  a lot of clippy warnings. I'll fix them

**Review by robin-nitrokey — APPROVED** [review]
LGTM!  We could even add `static_assertions` as a dependency.  It will probably come in handy in other crates too.

**Comment by sosthene-nitrokey** [comment]
I generally try to avoid bringing in entire dependencies for just 2 lines that will probably never need to be changed.
But it's true that for example `iso7816` also uses a similar macro.

**Pull request merged**